log/context.go

```diff
@@ -21,16 +22,28 @@ const (
 // All zero value fields will be ignored and only non-zero values will be
 // attached.
 type AttachArgs struct {
-	TraceID uint64
+	// The trace id of the span.
+	//
+	// If StringTraceID is set to true,
+	// it will be logged as string instead of int.
+	// The conversion to string will make it slightly slower,
+	// but avoids troubles when some of the log ingesters cannot deal with uint64
+	// numbers larger than max signed int64.
+	TraceID       uint64
+	StringTraceID bool
 }
 
 // Attach attaches a logger with data extracted from args into the context
 // object.
 func Attach(ctx context.Context, args AttachArgs) context.Context {
-	var kv []interface{}
+	kv := make([]interface{}, 0, 2)
```

Comment:
what's the reasoning behind this? Just that we're almost always going to have a TraceID so it's probably better to just pre-allocate this?

Reply:
I reverted this part back. In the current situation we at most do one allocation of 2 (because it's a single `append`, even we appended two items it won't be two allocations), but when there's no trace id set we can avoid this allocation. With `make` we always do the same allocation.